app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,269 @@
+
+            @BeforeEach
+            void prepare() {
+                givenTask = taskService.createTask(getTask());
+            }
+
+            @Test
+            @DisplayName("등록된 task 정보를 리턴한다.")
+            void it_return_task() {
+                Task foundTask = taskService.getTask(givenId());
+
+                assertThat(foundTask).isNotNull();
+            }
+        }
+
+        @Nested
+        @DisplayName("등록되지 않은 Task의 id 값이 주어진다면")
+        class Context_with_invalid_id {
+            @Test
+            @DisplayName("Task를 찾을 수 없다는 내용의 예외를 던진다.")
+            void it_return_taskNotFoundException() {
+                assertThatThrownBy(() -> taskService.getTask(INVALID_ID)).isInstanceOf(TaskNotFoundException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("등록할 Task가 주어진다면")
+        class Context_with_task {
+            Task givenTask;
+
+            @BeforeEach
+            void prepare() {
+                givenTask = getTask();
+            }
+
+            @Test
+            @DisplayName("Task를 생성하고, 리턴한다.")
+            void it_create_task_return_task() {
+                int sizeBeforeCreation = getTasksSize();
+
+                Task createdTask = taskService.createTask(givenTask);
+
+                int sizeAfterCreation = getTasksSize();
+
+                assertThat(createdTask.getTitle()).isEqualTo(givenTask.getTitle());
+                assertThat(sizeAfterCreation - sizeBeforeCreation).isEqualTo(1);
+            }
+        }
+
+        @Nested
+        @DisplayName("null 주어진다면")
+        class Context_without_task {
+            Task givenNullTask = null;
+
+            @Test
+            @DisplayName("NullPointerException을 던진다.")
+            void it_return_exception() {
+                assertThatThrownBy(() -> taskService.createTask(givenNullTask)).isInstanceOf(NullPointerException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("updateTask 메소드는")
+    class Describe_updateTask {
+        Task givenTask;
+        Task givenSource;
+        Long givenId() {
+            return givenTask.getId();
+        }
+
+        @BeforeEach
+        void prepare() {
+            givenTask = taskService.createTask(getTask());
+            givenSource = getTaskWithPostfix();
+        }
+
+        @Nested
+        @DisplayName("등록된 Task의 id 와 수정할 Task 가 주어진다면")
+        class Context_with_id_and_task {
+            @Test
+            @DisplayName("해당 id의 Task를 수정하고, 리턴한다.")
+            void it_update_task_return_task() {
+                Task updatedTask = taskService.updateTask(givenId(), givenSource);
+
+                assertThat(updatedTask.getTitle()).isEqualTo(givenSource.getTitle());
+            }
+        }
+
+        @Nested
+        @DisplayName("등록된 Task의 id 만 주어진다면")
+        class Context_with_id {
+            Task givenTask;
+            Task givenNullTask = null;
```

Comment:
`null`도 값입니다. `updateTask`에 주어져야 하는 두 개 파라미터 중 하나만 설명하고 있어요. 둘 다 설명해주는 것이 좋겠습니다. `id만 주어진다면`은 파라미터 하나만 주어지는 상황을 설명하는 것처럼 보여요.
```suggestion
        @Nested
        @DisplayName("등록된 Task의 id 와 null이 주어진다면")
        class Context_with_id {
            Task givenTask;
            Task givenNullTask = null;
```